flashlight/fl/tensor/backend/jit/JitBackend.cpp

```diff
@@ -111,253 +351,303 @@ Tensor JitBackend::fullWithType(const Shape& shape, T value, dtype type) {
   return jitTensorCreator_(ScalarNode::create(shape, type, value));
 }
 
-Tensor JitBackend::identity(const Dim /* dim */, const dtype /* type */) {
-  FL_JIT_BACKEND_UNIMPLEMENTED;
+Tensor JitBackend::identity(const Dim dim, const dtype type) {
+  return jitTensorCreator_(
+      ValueNode::create(wrappedBackend_.identity(dim, type)));
 }
 
-Tensor JitBackend::arange(
-    const Shape& /* shape */,
-    const Dim /* seqDim */,
-    const dtype /* type */) {
-  FL_JIT_BACKEND_UNIMPLEMENTED;
+Tensor
+JitBackend::arange(const Shape& shape, const Dim seqDim, const dtype type) {
+  return jitTensorCreator_(CustomNode::create(
+      "arange",
+      tensorsToNodes(),
+      Shape(shape),
+      [this, shape, seqDim, type](auto /* inputs */) {
+        return wrappedBackend_.arange(shape, seqDim, type);
+      }));
 }
 
-Tensor JitBackend::iota(
-    const Shape& /* dims */,
-    const Shape& /* tileDims */,
-    const dtype /* type */) {
-  FL_JIT_BACKEND_UNIMPLEMENTED;
+Tensor
+JitBackend::iota(const Shape& dims, const Shape& tileDims, const dtype type) {
+  return jitTensorCreator_(
+      ValueNode::create(wrappedBackend_.iota(dims, tileDims, type)));
 }
 
 /************************ Shaping and Indexing *************************/
-Tensor JitBackend::reshape(
-    const Tensor& /* tensor */,
-    const Shape& /* shape */) {
-  FL_JIT_BACKEND_UNIMPLEMENTED;
-}
-
-Tensor JitBackend::transpose(
-    const Tensor& /* tensor */,
-    const Shape& /* axes */ /* = {} */) {
-  FL_JIT_BACKEND_UNIMPLEMENTED;
-}
-
-Tensor JitBackend::tile(const Tensor& /* tensor */, const Shape& /* shape */) {
-  FL_JIT_BACKEND_UNIMPLEMENTED;
+Tensor JitBackend::reshape(const Tensor& tensor, const Shape& shape) {
+  if (tensor.shape().elements() != shape.elements()) {
+    std::ostringstream oss;
+    oss << "[JitBackend::reshape] Cannot reshape from " << tensor.shape()
+        << " to " << shape;
+    throw std::invalid_argument(oss.str());
+  }
+  return jitTensorCreator_(CustomNode::create(
+      "reshape",
+      tensorsToNodes(tensor),
+      Shape(shape),
+      [this, shape](auto inputs) {
+        return wrappedBackend_.reshape(*inputs.at(0), shape);
+      }));
+}
+
+Tensor JitBackend::transpose(const Tensor& tensor, const Shape& axes = {}) {
```

Comment:
Any plans to not use CustomNodes for these ops?

Possible my understanding of the existing `CustomNode` abstraction is wrong, but it seems like using a `CustonNode` for transpose precludes building generic passes that might affect it -- for example, removing an idempotent double transpose (`transpose(transpose(t))`. Possible motivation for a `UnaryOp`?

Reply:
> Any plans to not use CustomNodes for these ops?

Totally, I should've mentioned this.

Right now `CustumNode` serves 2 purposes:
1. Inject backend-specific optimization (OneDNN post-op fusion)
2. Cover ops we currently don't have optimization support for (i.e., `CustomNode` is opaque to optimization passes).

My goal is to keep IR minimal by following this workflow:
1. Identify hotspot via the upcoming GraphVizPrinter in workload we care about
2. Come up with optimization passes
3. Augment IR if needed, e.g., `TransposeNode` if we want to support `transepose(transpose(t)) --> t`

I want to make sure we don't add IR nodes & optimization passes for the sake of adding them -- they should be frequent or significant enough in the workload we wish to optimize.